File: BACKEND/src/main/java/com/macro/macro/Model/Microservico.java
package com.macro.macro.Model;

import com.macro.macro.Model.DTO.RequisicaoDTO;
import lombok.Getter;
import lombok.Setter;

import java.sql.Timestamp;
import java.util.List;


@Setter
@Getter
public class Microservico {

    private String id;
    private String nome;
    private String url;
    private List<RequisicaoDTO> requisicoes;
    private String descricao;

    private String status;
    private String healthEndpoint;
    private Long responseTime;

    private String createdAt;
    private String updatedAt;

}

File: FRONTEND/MC/src/app/model/Micro/microservico.ts
export interface RequisicaoDTO {
  endpoints: string;
  metodo: 'GET' | 'POST' | 'PUT' | 'DELETE';

  headers: any;

  queryParams: any;

  body: any;
}

export interface Microservico {
  id: string;
  nome: string;
  url: string;
  requisicoes: RequisicaoDTO[];
  descricao: string;

  status: string;
  healthEndpoint: string;
  responseTime: number;

  createdAt?: Date;
  updatedAt?: Date;
}